src/main/java/com/cgdpd/library/dto/BookCopy/BookCopyDTO.java

```diff
@@ -0,0 +1,24 @@
+package com.cgdpd.library.dto.BookCopy;
+
+import static com.cgdpd.library.validation.Validator.required;
+
+import com.cgdpd.library.entity.TrackingStatus;
+import java.util.Optional;
+import lombok.Builder;
+
+@Builder
+public record BookCopyDTO(Long id,
+                          Long bookId,
+                          TrackingStatus trackingStatus,
+                          Optional<Long> userId) {
+
+    public BookCopyDTO(Long id,
+                       Long bookId,
+                       TrackingStatus trackingStatus,
+                       Optional<Long> userId) {
+        this.id = required("id", id);
+        this.bookId = required("bookId", bookId);
+        this.trackingStatus = required("trackingStatus", trackingStatus);
+        this.userId = userId;
```

Comment:
Missing validations of user is there are situations when this is mandatory other when it can be present and others where it should be empty